xarm_api/src/xarm/debug/debug_print.cc

```diff
@@ -48,9 +48,10 @@ void print_nvect(const char *str, int vect[], int n) {
 }
 
 void print_hex(const char *str, unsigned char *hex, int len) {
-  char buf[len * 3 + 1] = {'\0'};
+  char *buf = new char[len * 3 + 1] {'\0'};

```

Comment:
This code looks like it could be massively simplified by just using printf in a loop.

Reply:
I was also thinking something similar but wanted to be 'minimally invasive' to start. Would be good to discuss it with uFactory when we make the PR.